File: app/src/main/java/org/scp/gymlog/room/daos/BarDao.java
package org.scp.gymlog.room.daos;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import org.scp.gymlog.room.entities.BarEntity;

import java.util.List;

@Dao
public interface BarDao {
    @Query("SELECT * FROM bar")
    List<BarEntity> getAll();

    @Insert
    long[] insertAll(BarEntity... bit);

    @Insert
    long insert(BarEntity bit);

    @Delete
    void delete(BarEntity bit);

    @Update
    void update(BarEntity bit);

    @Query("DELETE FROM bar")
    void clear();
}

File: app/src/main/java/org/scp/gymlog/room/daos/BarDao.kt
package org.scp.gymlog.room.daos

import androidx.room.*
import org.scp.gymlog.room.entities.BarEntity

@Dao
interface BarDao {
    @Query("SELECT * FROM bar")
    fun getAll(): List<BarEntity>

    @Insert
    fun insertAll(bar: List<BarEntity>): LongArray

    @Insert
    fun insert(bar: BarEntity): Long

    @Delete
    fun delete(bar: BarEntity)

    @Update
    fun update(bar: BarEntity)

    @Query("DELETE FROM bar")
    fun clear()
}
